model/map/src/main/java/org/keycloak/models/map/userSession/MapUserSessionProvider.java

```diff
@@ -81,7 +81,8 @@ public MapUserSessionProvider(KeycloakSession session, MapStorage<MapUserSession
     private Function<MapUserSessionEntity, UserSessionModel> userEntityToAdapterFunc(RealmModel realm) {
         // Clone entity before returning back, to avoid giving away a reference to the live object to the caller
         return (origEntity) -> {
-            if (origEntity.getExpiration() <= Time.currentTime()) {
+            long expiration = origEntity.getExpiration() != null ? origEntity.getExpiration() : 0l;
+            if (expiration <= Time.currentTime()) {
```

Comment:
Just out of curiosity, maybe we should start a discussion about this. Why are we doing this here? When we have the expiration field in store, should we just put this condition into MCB instead of manually checking it here?

Reply:
This covers every attempt to get a user/client session (also by `id` when MCB is not used) from the storage without a need to explicitly add it to MCB which can be easily forgotten.

Reply:
I created a discussion about this: https://github.com/keycloak/keycloak/discussions/11166